src/monitor/group_state_machine.c

```diff
@@ -881,6 +881,51 @@ ProceedGroupStateForPrimaryNode(AutoFailoverNode *primaryNode)
 		return true;
 	}
 
+	/*
+	 * when a secondary node has been removed during registration, or when
+	 * there's no visible reason to not be a primary rather than either
+	 * wait_primary or join_primary
+	 *
+	 *    join_primary ➜ primary
+	 */
+	if (IsCurrentState(primaryNode, REPLICATION_STATE_WAIT_PRIMARY) ||
```

Comment:
`REPLICATION_STATE_APPLY_SETTINGS` is another state that could transition into `PRIMARY`.

But my understanding is that adding `REPLICATION_STATE_APPLY_SETTINGS` would not be correct. Wanted to note in case my assessment is not accurate.

Reply:
Yeah that's because APPLY_SETTINGS is resolved by the primary itself reporting about it, whereas the other cases make progress depending on the state of the secondary nodes, which might fail to report, and we want to make progress when it makes sense to even when the secondaries are not helpful, or in the initial case, have been removed.